packages/trpc/server/routers/viewer/bookings/confirm.handler.ts

```diff
@@ -425,3 +412,48 @@ export const confirmHandler = async ({ ctx, input }: ConfirmOptions) => {
 
   return { message, status };
 };
+
+const checkIfUserIsAuthorizedToConfirmBooking = async ({
+  eventTypeId,
+  loggedInUserId,
+  teamId,
+  bookingUserId,
+  userRole,
+}: {
+  eventTypeId: number | null;
+  loggedInUserId: number;
+  teamId?: number | null;
+  bookingUserId: number | null;
+  userRole: string;
+}): Promise<void> => {
+  // check system wide admin
+  if (userRole === UserPermissionRole.ADMIN) return;
+
+  // Check if the user is the owner of the event type
+  if (bookingUserId === loggedInUserId) return;
+
+  // Check if user is associated with the event type
+  if (eventTypeId) {
+    const eventType = await prisma.eventType.findUnique({
+      where: {
+        id: eventTypeId,
+        OR: [{ hosts: { some: { userId: loggedInUserId } } }, { users: { some: { id: loggedInUserId } } }],
```

Comment:
I suspect this OR potentially results it in being a sequential search and we could definitely improve performance here even further. I would recommend running separate queries for this under `Promise All`. So something like `loggedInUserAsHostOfEventType` & `loggedInUserAsUserOfEventType` being the two separate queries, but under promise all.

Reply:
Okay. got it